src/Tensor.jl

```diff
@@ -70,7 +70,21 @@ Base.:(==)(a::Tensor, b::AbstractArray) = isequal(a, b)
 Base.:(==)(a::Tensor, b::Tensor) = isequal(a, b)
 Base.isequal(a::AbstractArray, b::Tensor) = false
 Base.isequal(a::Tensor, b::AbstractArray) = false
-Base.isequal(a::Tensor, b::Tensor) = allequal(labels.((a, b))) && allequal(parent.((a, b)))
+function Base.isequal(a::Tensor, b::Tensor)
+    issetequal(labels(a), labels(b)) || return false
+    perm = [findfirst(==(label), labels(b)) for label in labels(a)]
+    b_perm = permutedims(b, perm)
+    return allequal(labels.((a, b_perm))) && allequal(parent.((a, b_perm)))
```

Comment:
1. `allequal(labels.((a, b_perm)))` is redundant right?
2. `allequal` has another meaning... It's for checking that all elements in a collection are equal.
3. I prefer not to call `permutedims` on `b` but to call `eachindex` on `a` and permute the index.

```suggestion
    return all(eachindex(IndexCartesian(), a)) do I
        j = CartesianIndex(permute!(collect(Tuple(i)), perm))
        isequal(a[i], b[j])
    end
```